WebDriverAgentLib/Commands/FBDebugCommands.m

```diff
@@ -43,4 +47,25 @@ + (NSArray *)routes
   return FBResponseWithStatus(FBCommandStatusNoError, @{ @"tree": (accessibleTreeType ? application.fb_accessibilityTree : application.fb_tree) ?: @{} } );
 }
 
++ (id<FBResponsePayload>)handleGetSourceCommand:(FBRouteRequest *)request
+{
+  FBApplication *application = request.session.application ?: [FBApplication fb_activeApplication];
+  NSString *sourceType = request.parameters[@"type"];
+  id result;
+  FBCommandStatus status = FBCommandStatusNoError;
+  if ([sourceType caseInsensitiveCompare:@"json"] == NSOrderedSame) {
+    result = application.fb_tree;
+  } else if ([sourceType caseInsensitiveCompare:@"xml"] == NSOrderedSame) {
+    result = [FBXPath getSnapshotAsXMLString:application.lastSnapshot];
+  } else {
+    status = FBCommandStatusUnsupported;
+    result = [NSString stringWithFormat:@"Unknown source type '%@'. Only 'xml' and 'json' source types are supported.", sourceType];
+  }
+  if (!result) {
+    status = FBCommandStatusUnhandled;
```

Comment:
I would keep `FBCommandStatusUnsupported` & `FBCommandStatusUnhandled ` as fast returns otherwise it is hard to read.